clusterloader2/pkg/test/test.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+)
+
+// RunTest runs test based on provided test configuration.
+func RunTest(f *framework.Framework, conf *api.Config) error {
+	if f == nil {
+		return fmt.Errorf("framework must be provided")
+	}
+	if conf == nil {
+		return fmt.Errorf("test config must be provided")
+	}
+	defer f.DeleteAutomanagedNamespaces()
```

Comment:
We shouldn't be calling a deferred delete call before even creating. Please move this to after the "err != nil" block (so we don't try deleting un-created namespaces).

Reply:
But I still don't understand reason why you would ever want to call it even before creating? It doesn't make sense trying to delete N uncreated objects (just because it's not leading to any error) :)

Reply:
Maybe I wasn't clear earlier - I was suggesting moving this one line below instead of moving it after `if err != nil`

Reply:
> It doesn't make sense trying to delete N uncreated objects (just because it's not leading to any error) :)

It tries to delete only created ones.

> But I still don't understand reason why you would ever want to call it even before creating?

In this case it doesn't matter. Normally it depends on whether you want to execute code even if panic was risen or not. (assuming that I understand defer meaning correctly)

> Maybe I wasn't clear earlier - I was suggesting moving this one line below instead of moving it after if statement

This can be done :)

Reply:
> It tries to delete only created ones.

Doesn't matter even if the function is a no-op. What I'm saying here is "why should we make an unnecessary function call?"

> In this case it doesn't matter. Normally it depends on whether you want to execute code even if panic was risen or not. (assuming that I understand defer meaning correctly)

I'm not 100% sure, but defer wouldn't even be called if there's a panic before it - the binary crashes in such cases (unless panic handlers are registered).

> This can be done :)

Let's do that :)

Reply:
> Let's do that :)

Ok, it seems like you did make the change.